lib/rspec/rails/matchers/have_reported_error.rb

```diff
@@ -0,0 +1,193 @@
+        end
+
+        private
+
+        def error_matches_expectation?
+          return true if @expected_error.nil? && @error_subscriber.events.any?
+          return false if actual_error.nil?
+
+          case @expected_error
+          when Class
+            actual_error.is_a?(@expected_error)
+          when Exception
+            actual_error.is_a?(@expected_error.class) &&
+              (@expected_error.message.empty? || actual_error.message == @expected_error.message)
+          when Regexp
+            actual_error.message&.match(@expected_error)
+          when Symbol
+            actual_error == @expected_error
+          else
+            true
+          end
+        end
+
+        def attributes_match_if_specified?
+          return true if @attributes.empty?
+          return false if @error_subscriber.events.empty?
+
+          event_data = @error_subscriber.events.last[1]
+          attributes_match?(event_data)
+        end
+
+        def actual_error
+          @error_subscriber.events.empty? ? nil : @error_subscriber.events.last[0]
+        end
+
+        def attributes_match?(actual)
+          @attributes.all? do |key, value|
+            if defined?(RSpec::Matchers) && value.respond_to?(:matches?)
+              value.matches?(actual[key])
+            else
+              actual[key] == value
+            end
+          end
+        end
+
+        def unmatched_attributes(actual)
+          @attributes.reject do |key, value|
+            if defined?(RSpec::Matchers) && value.respond_to?(:matches?)
+              value.matches?(actual[key])
+            else
+              actual[key] == value
+            end
+          end
+        end
+      end
+
+      # @api public
+      # Passes if the block reports an error to `Rails.error`.
+      #
+      # This matcher asserts that ActiveSupport::ErrorReporter has received an error report.
+      #
+      # @example Checking for any error
+      #   expect { Rails.error.report(StandardError.new) }.to have_reported_error
+      #
+      # @example Checking for specific error class
+      #   expect { Rails.error.report(MyError.new) }.to have_reported_error(MyError)
+      #
+      # @example Checking for specific error instance with message
+      #   expect { Rails.error.report(MyError.new("message")) }.to have_reported_error(MyError.new("message"))
+      #
+      # @example Checking error attributes
+      #   expect { Rails.error.report(StandardError.new, context: "test") }.to have_reported_error.with(context: "test")
+      #
+      # @example Checking error message patterns
+      #   expect { Rails.error.report(StandardError.new("test message")) }.to have_reported_error(/test/)
+      #
+      # @example Negation
+      #   expect { "safe code" }.not_to have_reported_error
+      #
+      # @param expected_error [Class, Exception, Regexp, Symbol, nil] the expected error to match
+      def have_reported_error(expected_error = nil)
```

Comment:
We usually provide an alias to accommodate both styles, like ‘send_email’ and ‘have_sent_email’. Would you be opposed to have ‘reports_error’ as an alias?